### DISCUSSION TITLE
'isRequired' prop not working for me in my <Form> 's <TextField />

### DISCUSSION THREAD

**Discussion by sndp-s** [adobe/react-spectrum#4245] [Q&A] [answered]
I am not getting any errors when I submit the form empty, it submits the same way when there is no `isRequired` prop present, Please help.
My code sample
```react.js
    <Form
      isRequired
      necessityIndicator="icon"
      aria-label="Required with asterisk example"
      maxWidth="size-3600"
      onSubmit={onSubmit}
      validationState="invalid"
    >
      <TextField label="Name" />
      <ActionButton type="submit">Submit</ActionButton>
    </Form>
```
check it in [code sandbox](https://codesandbox.io/s/react-spectrum-textfield-isrequired-example-xldrz9)

**Comment by snowystinger** [comment, marked as answer]
The reason this is happening is because we use aria-required, not the required attribute. This was done as a result of this issue https://github.com/adobe/react-spectrum/issues/2123

The main point of which is, we don't want the native prompt. There may be form elements that are invisible because their UI is presented stylistically in a different way, such as ColorArea, ColorWheel, Slider, etc. So the native prompt would look strange.

And finally, there may be form elements which are not backed by a native input or are backed by several. These would need to be managed at the React State level anyways. Example, DateField, which consists of multiple segments.

If anything is required, you should prevent the form from submitting yourself and handle validation yourself. There are many libraries out there that will work with our hooks and components that you can leverage to make this simpler on yourself, such as https://react-hook-form.com/

**Reply by sndp-s** [reply]
Understood, Thanks a lot.
